### PR TITLE
Call BoxTreeRoot::construct from layout_thread_2020

### PR DESCRIPTION
**Pull request by nox**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/24360)

### CODE REVIEW

File: components/layout_thread_2020/dom_wrapper.rs [hunk 1]
```diff
@@ -108,6 +108,8 @@ pub struct ServoLayoutNode<'a> {
     /// Being chained to a PhantomData prevents `LayoutNode`s from escaping.
     chain: PhantomData<&'a ()>,
 }
+unsafe impl Send for ServoLayoutNode<'_> {}
+unsafe impl Sync for ServoLayoutNode<'_> {}
```
Comment (SimonSapin) [inline components/layout_thread_2020/dom_wrapper.rs]: I there a comment we could add here about the soundness of these impls?

File: components/layout_2020/dom_traversal.rs [hunk 2]
```diff
@@ -17,7 +17,7 @@ use style::properties::ComputedValues;
 use style::selector_parser::PseudoElement;
 
 #[derive(Clone, Copy)]
-pub(super) enum WhichPseudoElement {
+pub enum WhichPseudoElement {
```
Comment (SimonSapin) [inline components/layout_2020/dom_traversal.rs]: Are all of these newly-`pub` items used in another crate? I wonder if could move the crate boundaries somehow to make the API surface smaller. Maybe add some wrapper structs with private fields, for things that are exposed?

In any case this doesn’t need to block this PR.

**Review by SimonSapin — APPROVED** [review]

File: components/layout_2020/dom_traversal.rs (hunk 2 above)
Comment (nox) [reply components/layout_2020/dom_traversal.rs]: The trait `NodeExt` needs to be public, and thus any item exposed by the trait needs to be public too, so `WhichPseudoElement`, among other types, needs to be public. I could wrap LayoutBox into a struct but I'm not sure that it is *that* useful to do so.

File: components/layout_thread_2020/dom_wrapper.rs (hunk 1 above)
Comment (nox) [reply components/layout_thread_2020/dom_wrapper.rs]: I added a comment.

**Pull request merged**